llarp/router/router.cpp

```diff
@@ -666,52 +666,57 @@ namespace llarp
     BootstrapList b_list;
     for (const auto& router : configRouters)
     {
-      bool isListFile = false;
-      {
-        std::ifstream inf(router.c_str(), std::ios::binary);
-        if (inf.is_open())
-        {
-          const char ch = inf.get();
-          isListFile = ch == 'l';
-        }
-      }
-      if (isListFile)
-      {
-        if (not BDecodeReadFile(router, b_list))
-        {
-          throw std::runtime_error{fmt::format("failed to read bootstrap list file '{}'", router)};
-        }
-      }
-      else
-      {
-        RouterContact rc;
-        if (not rc.Read(router))
-        {
-          throw std::runtime_error{
-              fmt::format("failed to decode bootstrap RC, file='{}', rc={}", router, rc)};
-        }
-        b_list.insert(rc);
-      }
+      b_list.AddFromFile(router);
     }
 
     for (const auto& rc : conf.bootstrap.routers)
     {
       b_list.emplace(rc);
     }
 
-    for (auto& rc : b_list)
+    // in case someone has an old bootstrap file and is trying to use a bootstrap
+    // that no longer exists
+    for (auto rc_itr = b_list.begin(); rc_itr != b_list.end();)
     {
-      if (not rc.Verify(Now()))
+      if (rc_itr->IsObsoleteBootstrap())
+        b_list.erase(rc_itr);
+      else
+        rc_itr++;
+    }
+
+    auto verifyRCs = [&](bool ignoreObsolete) {
+      for (auto& rc : b_list)
       {
-        LogWarn("ignoring invalid RC: ", RouterID(rc.pubkey));
-        continue;
+        if (not rc.Verify(Now()))
+        {
+          LogWarn("ignoring invalid RC: ", RouterID(rc.pubkey));
+          continue;
+        }
+        if (ignoreObsolete and rc.IsObsoleteBootstrap())
```

Comment:
Should LogWarn here as well to let the admin know there is something being skipped in the bootstrap file.